### PR TITLE
Fix LogTarget to avoid creating folder

### PR DESCRIPTION
**Pull request by wapmorgan**
Fix `LogTarget` to avoid creating unnecessary folder in case when `php://stdout` or other `php://` log destination used.

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️/❌
| Fixed issues  | none

**Problem**:
When `php://stdout` used as log destination file, modified lines created `php:` folder in current folder.
This condition checks log destination and avoid this directory creation.

### CODE REVIEW

File: framework/log/FileTarget.php [hunk 1]
```diff
@@ -101,8 +101,10 @@ public function init()
      */
     public function export()
     {
-        $logPath = dirname($this->logFile);
-        FileHelper::createDirectory($logPath, $this->dirMode, true);
+        if (strncmp($this->logFile, 'php://', 6) !== 0) {
```
Comment (samdark) [inline framework/log/FileTarget.php]: How about other stream types?

**Comment by sup-ham** [comment]
I think FileTarget log should restricted to local file. you can create another eg. StdoutTarget if you want to use php://stdout and also STDOUT constant could be used instead as it always opened

**Comment by samdark** [comment]
Initially it was meant to work with files only...

**Comment by wapmorgan** [comment]
So, may be StreamTarget there would be a good solution?

**Comment by wapmorgan** [comment]
Also, don't want to create another issue.
Another propose is `JsonFileTarget` and `JsonStreamTarget`, that will format messages as json-encoded objects, describing log message, per line.

**Comment by samdark** [comment]
- JSON one looks way too specific to me to be part of the framework. 
- Are there any streams besides stdout that are usable as logger target?

**Comment by wapmorgan** [comment]
Zip, Gzip, custom streams. Also, `stderr` is an option.

**Comment by samdark** [comment]
Should we then check for these in `export()`?

**Comment by wapmorgan** [comment]
I propose check for `://` in `target`, this is not used in local files path.

**Comment by rob006** [comment]
> I propose check for `://` in `target`, this is not used in local files path.

`file://` is valid protocol, and it points to local filesystem.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍 Please add a line for CHANGELOG.md. Then I'll merge it.

**Comment by samdark** [comment]
Thank you 👍

**Pull request merged**